aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,291 @@
+	
+IHostingEnvironment (`source <https://github.com/aspnet/Hosting/blob/1.0.0-beta7/src/Microsoft.AspNet.Hosting.Abstractions/IHostingEnvironment.cs>`_)
+	Provides the current ``EnvironmentName``, ``WebRootPath``, and web root file provider. Available to the ``Startup`` constructor and ``Configure`` method. Learn more about :doc:`hosting`.
+	
+ILoggerFactory (`source <https://github.com/aspnet/Logging/blob/1.0.0-beta7/src/Microsoft.Framework.Logging.Abstractions/ILoggerFactory.cs>`_)
+	Provides a mechanism for creating loggers. Available to the ``Startup`` constructor and ``Configure`` method. Learn more about :doc:`logging`.
+	
+IServiceCollection (`source <https://github.com/aspnet/DependencyInjection/blob/1.0.0-beta7/src/Microsoft.Framework.DependencyInjection.Abstractions/IServiceCollection.cs>`_)
+	The current set of services configured in the container. Available only to the ``ConfigureServices`` method, and used by that method to configure the services available to an application.
+
+Looking at each method in the ``Startup`` class in the order in which they are called, the following services may be requested as parameters:
+
+Startup Constructor
+	- ``IApplicationEnvironment``
+	- ``IHostingEnvironment``
+	- ``ILoggerFactory``
+	
+ConfigureServices
+	- ``IServiceCollection``
+
+Configure
+	- ``IApplicationBuilder``
+	- ``IApplicationEnvironment``
+	- ``IHostingEnvironment``
+	- ``ILoggerFactory``
+	
+
+.. note:: Although ``ILoggerFactory`` is available in the constructor, it is typically configured in the ``Configure`` method. Learn more about the ``Startup`` class constructor and ``Configure`` method in :doc:`startup`.
+
+The ``ConfigureServices`` method is responsible for defining the services the application will use, including platform features like Entity Framework and ASP.NET MVC. Initially, the ``IServiceCollection`` provided to ``ConfigureServices`` has just a handful of services defined. The default web template shows an example of how to add additional services to the container using a number of extensions methods like ``AddEntityFramework``, ``Configure<Options>``, and ``AddMvc``.
+
+.. literalinclude:: ../../common/samples/WebApplication1/src/WebApplication1/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 45-84
+	:dedent: 8
+	:emphasize-lines: 5,11,18,24
+
+In order to keep the ``ConfigureServices`` method manageable, it is recommended that :doc:`middleware` and :doc:`request-features` authors provide extension methods and fluent configuration APIs that work with ``IServiceCollection``. The built-in features for Entity Framework, Identity, and ASP.NET MVC all follow this approach and can be used as models for providing a very succinct way to add a great deal of functionality to an ASP.NET application. You should also author your own middleware and request features to request and consume the services they need.
```

Comment:
See if there's a place later where this paragraph would fit. It's a recommendation for people writing a narrow type of ASP.NET extension and not something most need, especially in an intro to the `Startup` class.